app/src/models/cherry-pick.ts

```diff
@@ -8,3 +9,26 @@ export interface ICherryPickSnapshot {
   /** The progress of the operation */
   readonly progress: ICherryPickProgress
 }
+
+/** Union type representing the possible states of the cherry pick flow */
+export type CherryPickFlowStep = ChooseTargetBranchesStep
+
+export const enum CherryPickStep {
+  /**
+   * The initial state of a cherry pick.
+   *
+   * This is where the user picks which is the target of the cherry pick.
+   * This step will be skipped when cherry pick is initiated through
+   * drag and drop onto a specific branch.
+   */
+  ChooseTargetBranch = 'ChooseTargetBranch',
+}
+
+/** Shape of data needed to choose the base branch for a cherry pick  */
+export type ChooseTargetBranchesStep = {
+  readonly kind: CherryPickStep.ChooseTargetBranch
```

Comment:
Since `CherryPickStep` is used as a "kind" of step, maybe it should be renamed to `CherryPickStepKind` for clarity?
```suggestion
export const enum CherryPickStepKind {
  /**
   * The initial state of a cherry pick.
   *
   * This is where the user picks which is the target of the cherry pick.
   * This step will be skipped when cherry pick is initiated through
   * drag and drop onto a specific branch.
   */
  ChooseTargetBranch = 'ChooseTargetBranch',
}

/** Shape of data needed to choose the base branch for a cherry pick  */
export type ChooseTargetBranchesStep = {
  readonly kind: CherryPickStepKind.ChooseTargetBranch
```

Reply:
Makes sense. Thanks!